interface/views.py

```diff
@@ -114,12 +115,15 @@ def done(request, pk):
                                    pk=pk,
                                    score__isnull=True)
 
-    stdout = utils.decode(options['stdout']).split('\n')
+    stdout = utils.decode(options['stdout'])
     stderr = utils.decode(options['stderr'])
     exit_code = int(options['exit_code'])
 
-    submission.score = int(stdout[-2].split('/')[0])
-    submission.output = '\n'.join(stdout[:-2])
+    score = re.search(r'.*TOTAL: (\d+)/(\d+)', stdout, re.MULTILINE)
+    points = score.group(1) if score else 0
+
+    submission.score = points
+    submission.output = stdout + '\n' + stderr
```

Comment:
Eh, this is not quite correct, you should either capture stdout and stderr together so they interleave messages, or save them as separate fields. Concatenating them is impolite. If this is too much hassle for now, ok, but make an issue.

Reply:
issue #14